### PR TITLE
Update to Godot Engine 4.3 dev

### PR DESCRIPTION
**Pull request by fire**
Probably want to wait til 4.3 is released.

### CODE REVIEW

**Review by jonri — COMMENTED** [review]
@fire In reference to bd29413d1505a0b9ff489492a3b73af4f7db166a, I don't see any of the updated files mentioned in the description, and on the other hand it seems all of the copyright blocks have been changed to the boilerplate from Godot itself.

Regarding the other two, thanks for keeping things up with the upcoming 4.3 changes

File: SCsub [hunk 1]
```diff
@@ -4,16 +4,16 @@ Import("env")
...
-    "src/watercraft_ballast.cpp",
-    "src/watercraft_propulsion.cpp",
-    "src/watercraft_rudder.cpp",
+    "thirdparty/src/clippable_mesh.cpp",
+    "thirdparty/src/hydro_rigid_body.cpp",
+    "thirdparty/src/register_types.cpp",
+    "thirdparty/src/water_area.cpp",
+    "thirdparty/src/watercraft_ballast.cpp",
+    "thirdparty/src/watercraft_propulsion.cpp",
+    "thirdparty/src/watercraft_rudder.cpp",
```
Comment (jonri) [inline SCsub:13]: What advantage does moving these files bring?

File: register_types.h [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
 
-#include "src/register_types.h"
+#include "modules/hydro/thirdparty/src/register_types.h"
```
Comment (jonri) [inline register_types.h:31]: This appears to be incompatible with building the module using `custom_modules=...` with a directory not named `modules`.

File: SCsub (hunk 1 above)
Comment (fire) [reply SCsub:13]: I'm not sure how to disable the copyright header replacement.

File: register_types.h (hunk 2 above)
Comment (fire) [reply register_types.h:31]: Will investigate.

File: SCsub (hunk 1 above)
Comment (fire) [reply SCsub:13]: Another way is to add your copyright with SPDX short tag.

> GODOT ENGINE license boiler plate
>
> Copyright (c) 2019 Jon Ring
> SPDX-License-Identifier: MIT

Comment (fire) [reply SCsub:13]: And move it back.

**Comment by fire** [comment]
Will need to restart as I've forgotten all context.

**Pull request closed without merging**